library/x509write_crt.c

```diff
@@ -293,6 +293,44 @@ int mbedtls_x509write_crt_set_key_usage( mbedtls_x509write_cert *ctx,
     return( 0 );
 }
 
+int mbedtls_x509write_crt_set_ext_key_usage( mbedtls_x509write_cert *ctx,
+                                             const mbedtls_asn1_sequence *exts )
+{
+    unsigned char buf[256];
+    unsigned char *c = buf + sizeof(buf);
+    int ret;
+    size_t len = 0;
+    const mbedtls_asn1_sequence *last_ext = 0, *ext;
+
+    /* We need at least one extension: SEQUENCE SIZE (1..MAX) OF KeyPurposeId */
+    if( exts == NULL )
+        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+
+    /* Iterate over exts backwards, so we write them out in the requested order */
+    while( last_ext != exts )
+    {
+        for( ext = exts; ext->next != last_ext; ext = ext->next ) {}
```

Comment:
what do you think of the `continue;` idiom for this instead of empty braces?

```suggestion
        for( ext = exts; ext->next != last_ext; ext = ext->next ) continue;
```

Reply:
IMO marginally less readable. I couldn't find precedent either way.

Reply:
I'm not familiar with `continue;` as an idiom for an empty loop. I don't remember seeing an empty loop body in Mbed TLS. For empty if/else bodies, we use braces and a comment explaining why there's nothing to do in that case.

Reply:
I've seen it used in a few places, where actually having a keyword makes things easier to read, especially in codebases that don't require braces (e.g. in single-line `if`s or `for`s), and Mbed TLS is definitely one of those code bases.

e.g. https://man.openbsd.org/style.9 and search for `continue` gives

```
for (p = buf; *p != '\0'; ++p)
	continue;
```